diskutil.py

```diff
@@ -289,9 +289,9 @@ def getDiskBlockSize(dev):
         dev = '/dev/' + dev
     if isDeviceMapperNode(dev):
         return getDiskBlockSize(getDeviceSlaves(dev)[0])
-    if dev.startswith("/dev/"):
-        dev = re.match("/dev/(.*)", dev).group(1)
-    dev = dev.replace("/", "!")
```

Comment:
What specifically was wrong with the original code? The same pattern is used in several places in the installer (e.g. in `getDiskDeviceSize()` above) so if it is buggy then it should be fixed everywhere.

Reply:
Well for starters injecting `!` into a path is doomed to fail and using regexes instead of os primitives is rather reinventing the wheel.

Reply:
Apparently there are/were some devices having path separators in their name. And apparently the `!` was the `/sys` path encoding for these devices. `os` primitives and regexes are doing very different things, some are resolving symlinks converting to physical paths, the others are transforming the paths using just their names.

But @rosslagerwall point still remains, the same code is spread in different places so if one is broken why the others are fine?

I think adding `realpath` call before instead of replacing should work for both cases (your `/dev/by-id/...` and the `/dev/XXX/YYY`)

Reply:
And probably an exists check as well